fairscale/utils/meta.py

```diff
@@ -0,0 +1,306 @@
+####################################################################
+# BELOW: TAKEN FROM Pytorch-lightning. https://github.com/PyTorchLightning/pytorch-lightning/pull/9920#
+# Pruned part of the code dependent on Pytorch-lightning
+# TODO: Removed once merged and released on Pytorch-lightning side #
+####################################################################
+
+# Copyright The PyTorch Lightning team.
```

Comment:
We can use the FairScale copyright as long as we give credit to the PTL folks.

Reply:
I also remember there being some bug fixes so might be best if we take the latest code as opposed to something from the PR.